### ISSUE TITLE
lint,cmd/kfulint: record types info and parsed ast files inside context

### ISSUE THREAD

**Issue by Quasilyte** [PieselBois/kfulint#5]
Currently, `ast.ParseDir` results are ignored.
It returns multiple packages, usually it's target package (same name as containing directory name) only,
but sometimes there is also testing package with `_test` suffix.

Proposed plan:
1. Collect parsed packages.
2. Iterate over these packages, init context with current package files and types info.
3. Run checkers over that package with properly initialized context.
4. Repeat until there are no packages left.

**Note**: we may reject multi-package folders for now.
In other words: it's OK to exit if `len(packages) != 1` and that package name does not
match target package name. "proper" multi-package handling may be added later.
This can simplify initial implementation significantly.

`lint.Context` needs additional fields.
It definitely needs `*types.Info` and `[]*ast.File`. Maybe something else.

**Comment by fexolm** [comment]
Is it necessary for us to handle packages? Right now we don't have ( in plan ) any linters that would need any package information ( as far as i know ). So, maybe we would process every file independently?

**Comment by Quasilyte** [comment]
Yes, we can ignore that detail and operate on file level.

**Comment by Quasilyte** [comment]
#7 solved that.
